src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,198 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToInsert(index);
+        if (index == size) {
+            add(value);
+        } else {
+            linkBefore(value, findNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Node<T> antecedent = tail;
+        for (int i = 0; i < list.size(); i++) {
+            T value = list.get(i);
+            Node<T> newNode = new Node<>(antecedent, value, null);
+            if (antecedent == null) {
+                head = newNode;
+            } else {
+                antecedent.next = newNode;
+            }
+            antecedent = newNode;
+        }
+
+        tail = antecedent;
+        size += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexOfPresentElement(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexOfPresentElement(index);
+        Node<T> oldNode = findNode(index);
+        T oldValue = oldNode.item;
+        oldNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexOfPresentElement(index);
+        return unlink(findNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (node.item == null) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (object.equals(node.item)) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        }
         return false;
     }
```

Comment:
Don't forget that you can have `null` as a method parameter in the `remove(T object)` method. You have handled this correctly, so no action is needed.